src/coreclr/vm/method.cpp

```diff
@@ -3384,11 +3384,22 @@ BOOL NDirectMethodDesc::TryResolveNDirectTargetForNoGCTransition(_In_ MethodDesc
     }
     CONTRACTL_END
 
+    _ASSERTE(pMD->IsNDirect());
+    NDirectMethodDesc* pNMD = reinterpret_cast<NDirectMethodDesc*>(pMD);
+
+    if (!pMD->NDirectTargetIsImportThunk())
```

Comment:
The only bit that wasn't really clear to me after digging into this is whether `GetNativeNDirectTarget` needs to be taken into account.

It looked like that could be set for some condition and also expects `!NDirectTargetIsImportThunk()`

So should the below instead be `GetNativeNDirectTarget` which will return `m_pNativeNDirectTarget` if not null and `GetNDirectTarget` otherwise?

Reply:
`m_pNativeNDirectTarget` is dead code and it is always null. You can delete it and the `GetNativeNDirectTarget` method while you are on it.

Reply:
Thanks! Removed.